ext/dcos-installer/dcos_installer/action_lib/configure.py

```diff
@@ -44,6 +59,36 @@ def do_validate_gen_config(gen_config):
     return gen.validate(arguments=gen_config)
 
 
+def fetch_artifacts(gen_out) -> None:
+    """Returns None. Writes DC/OS generated templates and package artifacts to SERVE_DIR.
+
+    :param gen_out: gen.generate() output containing the keys 'artifacts' and 'packages'
+    :type  gen_out: dict | {}
+    """
+    try:
+        subprocess.check_output(['mkdir', '-p', SERVE_DIR])
+        for artifact in gen_out['artifacts']:
+            channel_path = artifact['channel_path']
+            local_content = artifact['local_content']
+            assert channel_path[-1] != '/', "Channel path looks like a directory and must be a file"
+            artifact_dst = SERVE_DIR + '/' + channel_path
+            log.debug('Writing artifact: {}'.format(artifact_dst))
+            subprocess.check_output(['mkdir', '-p', dirname(artifact_dst)])
+            with open(artifact_dst, 'w') as out:
+                out.write(local_content)
+        for pkg_id in gen_out['packages']:
```

Comment:
nit: this should be using the "copy out atomically" code we wrote for fetch_bootstrap. This could potentially leave half-written package files in the filesystem.